### PR TITLE
JAMES-2498 JDBCMailRepository should not materialize entities for counting them.

### PR DESCRIPTION
**Pull request by nstdio**
Executing SQL query for size.

### CODE REVIEW

File: server/data/data-jdbc/src/main/java/org/apache/james/mailrepository/jdbc/JDBCMailRepository.java [hunk 1]
```diff
@@ -721,6 +721,27 @@ protected void internalRemove(MailKey key) throws MessagingException {
         }
     }
 
+    @Override
+    public long size() throws MessagingException {
+        Connection conn = null;
```
Comment (mbaechler) [inline server/data/data-jdbc/src/main/java/org/apache/james/mailrepository/jdbc/JDBCMailRepository.java]: AFAIK all jdbc objects can by handled with a try-with-resource

Comment (nstdio) [reply server/data/data-jdbc/src/main/java/org/apache/james/mailrepository/jdbc/JDBCMailRepository.java]: Sure! I was trying to follow common pattern used in the class.

Comment (mbaechler) [reply server/data/data-jdbc/src/main/java/org/apache/james/mailrepository/jdbc/JDBCMailRepository.java]: In these cases, we often do a first commit to migrate code to the right code-style, then a second one with the change we want.

**Comment by chibenwa** [comment]
Hi,

This pull request has just been merged.

One very last thing: I don't have write access to this repository, hence I can not close this issue. Wouldn't you mind doing it for me? It would avoid me annoying the Apache INFRA team with such simple concerns.

Thanks again,

Cheers,

Benoit

**Pull request closed without merging**